### PR TITLE
docs: add getting started + updated contribution guide

### LINKED ISSUE #19: update docs
Update the docs on www.surrealdb.com (combine deno, node, browser into 1 page) also add bun runntime

### LINKED ISSUE #23: update repository/contributor docs
In https://github.com/surrealdb/surrealdb.js/issues/19 we discuss how to document the usage of this library.
In addition, we should also have some documentation for this repository. Here, I am thinking about

- [ ] repo structure
- [x] requirements
- [x] build/dev instructions
- [x] note on supported environments

I believe this is especially needed as we are mixing Deno, NodeJS, etc. in a project which is quite unusual.

### PR DESCRIPTION
**Pull request by mathe42**
fixes #23, fixes #19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,27 +2,146 @@
...
+just import it with
+
+```ts
+import Surreal from "surrealdb.js";
+```
+
+or when you use commonjs
+
+```ts
+const Surreal = require("surrealdb.js");
```
Comment (sasial-dev) [inline README.md:49]: Small code fix:
```suggestion
const Surreal = require("surrealdb.js").default;
```

File: README.md [hunk 2]
```diff
@@ -2,27 +2,146 @@
...
+
+```sh
+npm i surrealdb.js
+# or with yarn
+yarn add surrealdb.js
+# or with pnpm
+pnpm i surrealdb.js
+```
+
+just import it with
```
Comment (sasial-dev) [inline README.md:40]:
```suggestion
then, just import it with:
```

File: README.md [hunk 3]
```diff
@@ -2,27 +2,146 @@
...
+
+```ts
+import Surreal from "surrealdb.js";
+```
+
+or when you use commonjs
```
Comment (sasial-dev) [inline README.md:46]:
```suggestion
or when you use CommonJS
```

File: README.md [hunk 4]
```diff
@@ -2,27 +2,146 @@
...
+```ts
+import Surreal from "https://deno.land/x/surrealdb/mod.ts";
+```
+
+> Note you should set a version in the url! For example
+> `https://deno.land/x/surrealdb@0.5.0/mod.ts`
+
+#### NodeJS, Browser with bundler
+
+For NodeJS or Browser with bundler like rollup, vite, webpack, ... you can just
```
Comment (sasial-dev) [inline README.md:29]:
```suggestion
For NodeJS or a browser with bundler (for example: rollup, vite, or webpack) you can just:
```

File: README.md [hunk 5]
```diff
@@ -2,27 +2,146 @@
...
+Import it with
+
+```ts
+import Surreal from "https://deno.land/x/surrealdb/mod.ts";
+```
+
+> Note you should set a version in the url! For example
+> `https://deno.land/x/surrealdb@0.5.0/mod.ts`
+
+#### NodeJS, Browser with bundler
```
Comment (sasial-dev) [inline README.md:27]:
```suggestion
#### NodeJS, or browser with a bundler
```

File: README.md [hunk 6]
```diff
@@ -2,27 +2,146 @@
...
 #### Supported environments
-* [Deno](https://deno.land)
-* [NodeJS](https://nodejs.org)
-* [Bun](https://bun.sh)
-* web-browsers
+
+- [Deno](https://deno.land)
+- [NodeJS](https://nodejs.org)
+- [Bun](https://bun.sh)
+- web-browsers
```
Comment (sasial-dev) [inline README.md:136]:
```suggestion
- Web Browsers
```

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Some grammar fixes + a code fix for the CommonJS import!

File: README.md (hunk 1 above)
Comment (kratess) [reply README.md]: This helped through my struggle to make this thing work. Thanks

File: README.md [hunk 7]
```diff
@@ -2,27 +2,146 @@
...
+then, just import it with:
+
+```ts
+import Surreal from "surrealdb.js";
+```
+
+or when you use CommonJS
+
+```ts
+const Surreal = require("surrealdb.js").default;
```
Comment (kearfy) [inline README.md:49]:
```suggestion
const { default: Surreal } = require("surrealdb.js");
```

**Review by kearfy — CHANGES_REQUESTED** [review]

**Review by kearfy — APPROVED** [review]

**Pull request merged**